### PR TITLE
python3Packages.tpm2-pytss: add missing pyyaml dependency

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Description of changes

Fix the build

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by pyrox0 — APPROVED** [review]
LGTM!

**Comment by theHedgehog0** [comment]
Result of `nixpkgs-review pr 184684` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>tpm2-pkcs11</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.tpm2-pytss</li>
    <li>python39Packages.tpm2-pytss</li>
  </ul>
</details>

File: pkgs/development/python-modules/tpm2-pytss/default.nix [hunk 1]
```diff
@@ -27,9 +28,8 @@ buildPythonPackage rec {
 
   nativeBuildInputs = [
     cffi
-    pkgconfig
-    # somehow propagating from pkgconfig does not work
-    pkg-config
+    pkgconfig # this is the python module
+    pkg-config # this is the actual pkg-config tool
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/tpm2-pytss/default.nix:32]: pkgconfig should actually contain pkg-config already or are they shelling out? Also we don't really need those comments but whatever.

Comment (peterhoeg) [reply pkgs/development/python-modules/tpm2-pytss/default.nix:32]: > pkgconfig should actually contain pkg-config already or are they shelling out? 

No idea what they are doing, but having both makes it work (this was also the case previously).

> Also we don't really need those comments but whatever.

Oh, we absolutely do. If anything we need a lot *more* comments in nixpkgs in general. Whenever anything unusual is done, we should definitely add  a comment.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/tpm2-pytss/default.nix:32]: > No idea what they are doing, but having both makes it work (this was also the case previously).

pkgconfig should propagate pkg-config into nativeBuildInputs. That only works sometimes and I gave up debugging why it not always works.

**Pull request merged**